packages/table/src/Table.ts

```diff
@@ -0,0 +1,463 @@
+        const { target } = event;
+        const { parentElement: rowItem } = target as HTMLElement & {
+            parentElement: TableRow;
+        };
+        if (!rowItem.value) {
+            const { checkbox } = target as TableCheckboxCell;
+            if (checkbox.checked || checkbox.indeterminate) {
+                this.selectAllRows();
+            } else {
+                this.deselectAllRows();
+            }
+        } else {
+            switch (this.selects) {
+                case 'single': {
+                    this.deselectAllRows();
+                    if (rowItem.selected) {
+                        this.selectedSet.add(rowItem.value);
+                        this.selected = [...this.selectedSet];
+                    }
+                    break;
+                    // if you have multiple selected, but selects='single',
+                    // we deselect all cells? hmm...
+                    // EXAMPLE: rows 1, 2, 4 are selected but selects is single.
+                    // we deselect 2 => 1 and 4 are STILL SELECTED. if you want
+                    // to select any item, it's going to deselect all the cells before
+                    // reselecting the cell you just clicked on, as seen above.
+                }
+                case 'multiple': {
+                    if (rowItem.selected) {
+                        this.selectedSet.add(rowItem.value);
+                    } else {
+                        this.selectedSet.delete(rowItem.value);
+                    }
+                    this.selected = [...this.selectedSet];
+
+                    const tableRows = [
+                        ...this.querySelectorAll('sp-table-row'),
+                    ] as TableRow[];
+
+                    const allSelected =
+                        this.selected.length === tableRows.length;
+
+                    if (!this.tableHeadCheckboxCell) return;
+                    this.tableHeadCheckboxCell.checked = allSelected;
+                    this.tableHeadCheckboxCell.indeterminate =
+                        this.selected.length > 0 && !allSelected;
+                    break;
+                }
+                default: {
+                    break;
+                }
+            }
+        }
+        event.stopPropagation();
+        this.dispatchEvent(
+            new Event('change', {
+                cancelable: true,
+                bubbles: true,
+                composed: true,
+            })
+        );
+    }
+
+    public scrollToIndex(index?: number): void {
+        if (index) {
+            this.renderVirtualizedItems(index);
+        }
+    }
+
+    protected override render(): TemplateResult {
+        return html`
+            <slot @change=${this.handleChange}></slot>
+        `;
+    }
+
+    protected override willUpdate(changed: PropertyValues<this>): void {
+        if (!this.hasUpdated) {
+            const rowValues = new Set<string>();
```

Comment:
Might be worth brining this into an `initialize` method or something so that it's easier to differentiate from repeated life cyle work.